sdks/java/io/solr/src/test/java/org/apache/beam/sdk/io/solr/SolrIOIT.java

```diff
@@ -0,0 +1,174 @@
+      }
+    } else {
+      LOG.info("Create and populate collection {}",
+          READ_COLLECTION);
+    }
+
+    SolrIOTestUtils.createCollection(WRITE_COLLECTION, 2, 2, solrClient);
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    SolrIOTestUtils.deleteCollection(WRITE_COLLECTION, solrClient);
+    solrClient.close();
+  }
+
+  @Test
+  public void testSplitsVolume() throws Exception {
+    SolrIO.Read read =
+            SolrIO.read().withConnectionConfiguration(readConnectionConfiguration);
+    SolrIO.BoundedSolrSource initialSource =
+        new SolrIO.BoundedSolrSource(read, null);
+    //desiredBundleSize is ignored now
+    long desiredBundleSizeBytes = 0;
+    List<? extends BoundedSource<SolrDocument>> splits =
+        initialSource.split(desiredBundleSizeBytes, options);
+    SourceTestUtils.assertSourcesEqualReferenceSource(initialSource, splits, options);
+    //this is the number of Solr shards
+    long expectedNumSplits = NUM_SHARDS;
+    assertEquals(expectedNumSplits, splits.size());
+    int nonEmptySplits = 0;
+    for (BoundedSource<SolrDocument> subSource : splits) {
+      if (readFromSource(subSource, options).size() > 0) {
+        nonEmptySplits += 1;
+      }
+    }
+    assertEquals(expectedNumSplits, nonEmptySplits);
+  }
+
+  @Test
+  public void testReadVolume() throws Exception {
+    PCollection<SolrDocument> output =
+        pipeline.apply(
+            SolrIO.read().withConnectionConfiguration(readConnectionConfiguration));
+    PAssert.thatSingleton(output.apply("Count", Count.<SolrDocument>globally()))
+        .isEqualTo(NUM_DOCS);
+    pipeline.run();
+  }
+
+  @Test
+  public void testWriteVolume() throws Exception {
+    SolrIO.ConnectionConfiguration writeConnectionConfiguration =
+        SolrIO.ConnectionConfiguration.create(options.getSolrZookeeperServer(), WRITE_COLLECTION);
+    List<SolrInputDocument> data = SolrIOTestUtils.createDocuments(NUM_DOCS);
+    pipeline
+        .apply(Create.of(data))
+        .apply(SolrIO.write().withConnectionConfiguration(writeConnectionConfiguration));
+    pipeline.run();
+
+    long currentNumDocs = SolrIOTestUtils.commitAndGetCurrentNumDocs(
+        WRITE_COLLECTION, solrClient);
+    assertEquals(NUM_DOCS, currentNumDocs);
+  }
+
+  @Test
+  public void testEstimatedSizesVolume() throws Exception {
+    SolrIO.Read read =
+        SolrIO.read().withConnectionConfiguration(readConnectionConfiguration);
+    SolrIO.BoundedSolrSource initialSource =
+        new SolrIO.BoundedSolrSource(read, null);
+    // can't use equal assert as Solr collections never have same size
+    // (due to internal Lucene implementation)
+    long estimatedSize = initialSource.getEstimatedSizeBytes(options);
+    LOG.info("Estimated size: {}", estimatedSize);
+    assertThat(
+        "Wrong estimated size bellow minimum",
+        estimatedSize,
+        greaterThan(SolrIOTestUtils.AVERAGE_DOC_SIZE * NUM_DOCS));
```

Comment:
Perhaps AVERAGE_DOC_SIZE should be really MINIMUM_DOC_SIZE? Cause otherwise, an ideal implementation would return estimatedSize == average * num, by definition of average.